File: app/models/question.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field
from pydantic import ConfigDict


class QuestionBase(BaseModel):
    type: str = Field(pattern=r"^T[23]$")
    source: str
    year: int
    month: int
    suite: Optional[str] = None
    number: Optional[str] = None
    title: str
    body: str


class QuestionCreate(QuestionBase):
    tags: List[str] = Field(default_factory=list)


class QuestionUpdate(BaseModel):
    title: Optional[str] = None
    tags: Optional[List[str]] = None


class QuestionRead(QuestionBase):
    id: int
    slug: Optional[str] = None
    tags: List[str] = Field(default_factory=list)
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/question.ts
export interface DirectionPlanItem {
  title: string;
  summary?: string;
  stance?: string | null;
  structure?: string[];
}

export interface DirectionPlan {
  recommended?: DirectionPlanItem;
  alternatives?: DirectionPlanItem[];
}

export interface QuestionPayload {
  type: 'T2' | 'T3';
  source: string;
  year: number;
  month: number;
  suite?: string | null;
  number?: string | null;
  title: string;
  body: string;
  tags: string[];
  direction_plan?: DirectionPlan | null;
}

export interface Question extends QuestionPayload {
  id: number;
  slug?: string | null;
  created_at: string;
  updated_at: string;
}

export interface FetchTask {
  id: number;
  type: string;
  status: string;
  session_id: number | null;
  answer_id: number | null;
  payload: Record<string, unknown>;
  result_summary: Record<string, unknown>;
  error_message?: string | null;
  created_at: string;
  updated_at: string;
}

export interface FetchQuestionResult {
  slug: string;
  title: string;
  type: string;
  year: number;
  month: number;
  suite?: string;
  number?: string;
  body: string;
  source_url: string;
}
